contrib/codegen/templates/input/input_snapshot.gotmpl

```diff
@@ -201,47 +214,38 @@ type BuildOptions struct {
 {{- $kindLowerCamel := lower_camel $resource.Kind }}
 {{- $kindLowerCamelPlural := pluralize $kindLowerCamel }}
     // List options for composing a snapshot from {{ $kindPlural }}
-    {{ $kindPlural }} []client.ListOption
+    {{ $kindPlural }} ResourceBuildOptions
 {{- end }}
 {{- end }}
 }
 
+// Options for reading resources of a given type
+type ResourceBuildOptions struct {
+{{/* generate build options here */}}
+    // List options for composing a snapshot from a resource type
+    ListOptions []client.ListOption
+
+    // If provided, ensure the resource has been verified before adding it to snapshots
+    Verifier verifier.ServerResourceVerifier
+}
+
```

Comment:
Any reason to generate this many times rather than have it live one place?

Reply:
it makes it easier for the user to write, by only needing to import one package